File: grove-temperature-humidity-sensor-sht3x/grove-temperature-humidity-sensor-sht3x.py
#!/usr/bin/env python

import os # add for node-red
import sys # add for node-red
import json # add  for node-red
import time

from grove.grove_temperature_humidity_sensor_sht3x import GroveTemperatureHumiditySensorSHT3x

def main():
    # get address
    arguments = sys.argv

    # print disable
    # sys.stdout = open(os.devnull, 'w')

    sensor = GroveTemperatureHumiditySensorSHT3x()

    # print enable
    # sys.stdout = sys.__stdout__

    temperature, humidity = sensor.read()
    print(json.dumps({"temperature":'{:.2f}'.format(temperature), "humidity":'{:.2f}'.format(humidity)}))


if __name__ == "__main__":
    main()

File: grove-temperature-humidity-sensor-sht3x/grove-temperature-humidity-sensor-sht3x.js
var path = require('path');

module.exports = function(RED) {
    function GroveTemperatureHumiditySensorSHT3x(config) {
        RED.nodes.createNode(this,config);
        
        this.port_name = "I2C";
        this.status({fill:"blue",shape:"dot",text:this.port_name});

        var node = this;
        node.on('input', function(msg) {
            const exec = require('child_process').exec;
            this.status({fill:"yellow",shape:"ring",text:this.port_name + " connecting"});
            
            try {
                exec('python -u ' + path.join( __dirname , 'grove-temperature-humidity-sensor-sht3x.py' ) + ' ' + config.model, (err, stdout, stderr) => {
                    if (err) { console.log(err); }
                    console.log(stdout);
                    msg.payload = JSON.parse(stdout);
                    msg.payload.temperature = Number(msg.payload.temperature);
                    msg.payload.humidity = Number(msg.payload.humidity);
                    this.status({fill:"green",shape:"dot",text:this.port_name + " connected"});
                    node.send(msg);
                });
            } catch (e){
                console.error(e);
            }
            
        });
    }
    RED.nodes.registerType("grove-temperature-humidity-sensor-sht3x",GroveTemperatureHumiditySensorSHT3x);
}
